marlowe-runtime/cip30-demo/Main.hs

```diff
@@ -54,6 +83,70 @@ getWalletR walletName = Yesod.defaultLayout do
         <i>Withdraw payouts from a running contract
   |]
 
+getWalletCreateR :: WalletName -> Handler Yesod.Html
+getWalletCreateR walletName = Yesod.defaultLayout do
+  Yesod.toWidget [Yesod.hamlet|
+    <input id="contractSource" type="text">
+    <button id="postContractSource">click
+    <script>
+      const contractSource = document.getElementById('contractSource')
+      const postContractSource = document.getElementById('postContractSource')
+      postContractSource.addEventListener("click", async () => {
+        const options = {
+          method: "POST",
+          headers: { "Content-Type": "application/json;charset=utf-8" },
+          body: JSON.stringify({ source: contractSource.value })
```

Comment:
As mentioned above, let's parse the contract source here into JSON

```suggestion
          body: JSON.stringify({ source: JSON.parse(contractSource.value) })
```

Reply:
Do we know that the source will be given in JSON?